### ISSUE TITLE
"ReadFileHandle.Read error" occured when reading files

### ISSUE THREAD

**Issue by CURAS** [rclone/rclone#2219]
When I start `rclone 1.39` on `openSUSE leap 42.3` by `./rclone mount onedrive: ./onedrive --config rclone.conf --allow-other -vv`, I can use `echo test > test.txt` to write a file to my OneDrive storage, and I can read this file by `cat test.txt`. But when I edit this file and save it on my OneDrive client which is built in Windows 10, I can't read it again and disappeared the following messages.
![d](https://user-images.githubusercontent.com/14164381/38412764-a239cc0e-39bc-11e8-8f20-1259b70b4196.png)

**Comment by ncw** [comment]
Can you try rclone v1.40 or even better [the latest beta](https://beta.rclone.org)?

I'm hoping this is fixed already!

It is possible you might have to add `--vfs-cache-mode writes` though.

**Comment by CURAS** [comment]
Even though I added `--vfs-cache-mode writes` and try the latest beta version, this problem remains again.

**Comment by ncw** [comment]
@CURAS can you give me exact instructions to reproduce the problem please?  What exactly did you do on Windows?  Which editor did you use?

**Comment by CURAS** [comment]
I don't know what you mean exactly, so I can only tell you what I did.

First, I mounted my OneDrive storage on my computer by rclone, and created a file on this directory. Then I can read and write on this file.

Second, I opened my another computer with Windows 10 installed, which had set to keep it synchronized with my OneDrive storage before. So I can read and write on the file on last step.

Third, after changing content of the file on this computer, I came back to the first computer and tried to read what I wrote, I could only get the message just like the first picture.


I think it should be irrelevant to editor, because I tried `Notepad`, `Visual Studio Code` and some of other editors, and got the same result.
